### PR TITLE
Theme selector setting

### PR DESCRIPTION
**Pull request by YodaForce157**

### CODE REVIEW

**Comment by YodaForce157** [comment]
About to fix major issue, give me a minute

**Comment by YodaForce157** [comment]
Fixed.

File: src/main/java/io/github/railroad/Railroad.java [hunk 1]
```diff
@@ -46,8 +47,23 @@ public static Stage getWindow() {
         return window;
     }
 
+    public static void setStyle(String theme) {
+        getScene().getStylesheets().clear();
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/Railroad.java]: This is not great as a long term solution as it clears all stylesheets even if they're not necessarily themes. Instead we should consider having some kind of container holding what the current theme is and that way we can remove that specific theme from the sheets. Possibly consider a global atomic reference maybe?

File: src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java [hunk 2]
```diff
@@ -1,6 +1,71 @@
...
+
+        var themeBox = new RRVBox();
+
+        title.setStyle("-fx-font-size: 20pt; -fx-font-weight: bold;");
+        title.prefWidthProperty().bind(widthProperty());
+        title.setAlignment(Pos.CENTER);
+
+        themeBox.setStyle("-fx-padding: 0 0 0 10");
+
+        var themesDir = new File("themes").listFiles();
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java]: I would prefer that we avoid using java.io where possible and try to use nio. As for how you'd do that here, probably `Files.listFiles(Path)` which gives you a `Steam<Path>`.

File: src/main/java/io/github/railroad/Railroad.java [hunk 3]
```diff
@@ -26,13 +26,15 @@
...
 import java.util.Optional;
+import java.util.concurrent.atomic.AtomicReference;
 import java.util.function.Consumer;
 
 public class Railroad extends Application {
     public static final OkHttpClient HTTP_CLIENT = new OkHttpClient();
     public static final Gson GSON = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
     public static final ProjectManager PROJECT_MANAGER = new ProjectManager();
     public static final PluginManager PLUGIN_MANAGER = new PluginManager();
+    public static AtomicReference<String> THEME = new AtomicReference<>("default-dark");
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/Railroad.java]:
```suggestion
    public static final AtomicReference<String> THEME = new AtomicReference<>("default-dark");
```

File: src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java [hunk 4]
```diff
@@ -28,15 +28,22 @@ public SettingsAppearancePane() {
...
 
-        var themesDir = new File("themes").listFiles();
+        File[] themes = new File[0];
+        try {
+             themes = Files.list(new File("themes").toPath())
+                    .filter(file -> file.toString().endsWith(".css"))
+                    .map(file -> file.toFile()).toArray(File[]::new);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java]:
```suggestion
        List<Path> themes = new ArrayList<>();
        try(Stream<Path> files = Files.list(Path.of("themes"))) {
             files.filter(file -> file.endsWith(".css")).forEach(themes::add);
        } catch (IOException e) {
            e.printStackTrace();
        }
```

File: src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java [hunk 5]
```diff
@@ -28,15 +28,22 @@ public SettingsAppearancePane() {
...
 
-        if(themeSelector.getItems().size() < themesDir.length){
+        if(themeSelector.getItems().size() < themes.length + 2){
             themeSelector.getItems().clear();
 
             themeSelector.getItems().addAll("default-dark", "default-light");
 
-            for (var theme : Arrays.stream(themesDir).filter(file -> file.getName().endsWith(".css")).toArray(File[]::new)) {
+            for (var theme : themes) {
                 String name = theme.getName().replace(".css", "");
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java]:
```suggestion
                String name = theme.getFileName().toString().replace(".css", "");
```

File: src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java [hunk 6]
```diff
@@ -28,15 +28,22 @@ public SettingsAppearancePane() {
...
+             themes = Files.list(new File("themes").toPath())
+                    .filter(file -> file.toString().endsWith(".css"))
+                    .map(file -> file.toFile()).toArray(File[]::new);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
         themeSelector.setPrefWidth(180);
 
-        if(themeSelector.getItems().size() < themesDir.length){
+        if(themeSelector.getItems().size() < themes.length + 2){
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java]:
```suggestion
        if(themeSelector.getItems().size() < themes.size() + 2) {
```

File: src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java [hunk 7]
```diff
@@ -28,15 +28,22 @@ public SettingsAppearancePane() {
...
             themeSelector.getItems().clear();
 
             themeSelector.getItems().addAll("default-dark", "default-light");
 
-            for (var theme : Arrays.stream(themesDir).filter(file -> file.getName().endsWith(".css")).toArray(File[]::new)) {
+            for (var theme : themes) {
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/railroad/settings/ui/SettingsAppearancePane.java]:
```suggestion
            for (Path theme : themes) {
```

**Review by DaRealTurtyWurty — CHANGES_REQUESTED** [review]

**Review by DaRealTurtyWurty — APPROVED** [review]

**Pull request merged**